src/pgrepr/src/value.rs

```diff
@@ -115,15 +122,48 @@ impl Value {
                 buf.push_row(jsonb.into_row()).unpack_first(),
                 ScalarType::Jsonb,
             ),
+            Value::List(elems) => {
+                let typed_elems = elems
+                    .into_iter()
+                    .map(|elem| {
+                        elem.map_or((Datum::Null, ScalarType::Unknown), |elem| {
+                            elem.into_datum(buf)
+                        })
+                    })
+                    .collect::<Vec<_>>();
+                let mut elem_type = ScalarType::Unknown;
```

Comment:
This doesn't seem like the correct default for an empty list? I think this function isn't getting called right now because you haven't implemented the decode side of things, so this logic isn't getting exercised AFAICT.

Reply:
What should the type of an empty list be?

Actually, backing up, when do we decode a value without knowing the intended type? Is there a ScalarType somewhere that we can pass in here?

Reply:
In sql queries, I've copied the behaviour from arrays where you can't construct an empty array unless it has a type hint.

```
jamie=# select array[];
ERROR:  cannot determine type of empty array
LINE 1: select array[];
               ^
HINT:  Explicitly cast to the desired type, for example ARRAY[]::integer[].
jamie=# select array[] :: int array;
 array 
-------
 {}
(1 row)
```

Reply:
I think a pgrepr::Type, but yeah I think it’s around.

Reply:
I mean it's sql, so maybe we just silently default to `TEXT`.